### PR TITLE
feat: export ersi and algolia result types

### PR DESCRIPTION
**Pull request by franciscofsales**
Added exports to types that are commonly used outside of the lib.

### CODE REVIEW

**Comment by franciscofsales** [comment]
Possibly, I personally am using esri, but noticed algolia types weren't being exported either

**Comment by mtmail** [comment]
I created a PR to remove Algolia https://github.com/smeijer/leaflet-geosearch/pull/402

**Review by smeijer — APPROVED** [review]

**Comment by smeijer** [comment]
Thank you!

File: src/providers/algoliaProvider.ts [hunk 1]
```diff
@@ -86,21 +86,21 @@ export default class Provider extends AbstractProvider<
     return [
       // Building + Street
       result.locale_names?.default[
-        this.findBestMatchLevelIndex(
-          result._highlightResult.locale_names.default,
-        )
+      this.findBestMatchLevelIndex(
```
Comment (smeijer) [inline src/providers/algoliaProvider.ts:89]:
```suggestion
        this.findBestMatchLevelIndex(
```

File: src/providers/algoliaProvider.ts [hunk 2]
```diff
@@ -86,21 +86,21 @@ export default class Provider extends AbstractProvider<
...
+        result._highlightResult.locale_names.default,
```
Comment (smeijer) [inline src/providers/algoliaProvider.ts:90]:
```suggestion
          result._highlightResult.locale_names.default,
```

File: src/providers/algoliaProvider.ts [hunk 3]
```diff
@@ -86,21 +86,21 @@ export default class Provider extends AbstractProvider<
...
+      )
```
Comment (smeijer) [inline src/providers/algoliaProvider.ts:91]:
```suggestion
        )
```

File: src/providers/algoliaProvider.ts [hunk 4]
```diff
@@ -86,21 +86,21 @@ export default class Provider extends AbstractProvider<
...
       ],
       // City
       result.city?.default[
-        this.findBestMatchLevelIndex(result._highlightResult.city.default)
+      this.findBestMatchLevelIndex(result._highlightResult.city.default)
```
Comment (smeijer) [inline src/providers/algoliaProvider.ts:95]:
```suggestion
        this.findBestMatchLevelIndex(result._highlightResult.city.default)
```

File: src/providers/algoliaProvider.ts [hunk 5]
```diff
@@ -86,21 +86,21 @@ export default class Provider extends AbstractProvider<
...
       ],
       // Administrative (State / Province)
       result.administrative[
-        this.findBestMatchLevelIndex(result._highlightResult.administrative)
+      this.findBestMatchLevelIndex(result._highlightResult.administrative)
```
Comment (smeijer) [inline src/providers/algoliaProvider.ts:99]:
```suggestion
        this.findBestMatchLevelIndex(result._highlightResult.administrative)
```

File: src/providers/algoliaProvider.ts [hunk 6]
```diff
@@ -86,21 +86,21 @@ export default class Provider extends AbstractProvider<
...
       ],
       // Zip code / Postal code
       result.postcode?.[
-        this.findBestMatchLevelIndex(result._highlightResult.postcode)
+      this.findBestMatchLevelIndex(result._highlightResult.postcode)
```
Comment (smeijer) [inline src/providers/algoliaProvider.ts:103]:
```suggestion
        this.findBestMatchLevelIndex(result._highlightResult.postcode)
```

**Comment by smeijer** [comment]
Thanks again! I've also added you as a collaborator on the project. Please continue to make PRs as you feel the need. You can now make your branches directly on the repo rather than your fork if you want.

Thanks for your contributions! Much appreciated.

**Comment by smeijer** [comment]
@all-contributors please add @franciscofsales for code

**Pull request merged**